cloudinit/config/cc_locale.py

```diff
@@ -6,30 +6,46 @@
 #
 # This file is part of cloud-init. See LICENSE file for license information.
 
-"""
-Locale
-------
-**Summary:** set system locale
+"""Locale: set system locale"""
 
-Configure the system locale and apply it system wide. By default use the locale
-specified by the datasource.
-
-**Internal name:** ``cc_locale``
-
-**Module frequency:** per instance
-
-**Supported distros:** all
-
-**Config keys**::
-
-    locale: <locale str>
-    locale_configfile: <path to locale config file>
-"""
+from textwrap import dedent
 
 from cloudinit import util
+from cloudinit.config.schema import get_schema_doc, validate_cloudconfig_schema
+from cloudinit.settings import PER_INSTANCE
+
+
+frequency = PER_INSTANCE
+distros = ['all']
+schema = {
+    'id': 'cc_locale',
+    'name': 'Locale',
+    'title': 'Set system locale',
+    'description': dedent("""\
+        Configure the system locale and apply it system wide. By default use
+        the locale specified by the datasource."""),
+    'distros': distros,
+    'examples': ["TODO"],
+    'frequency': frequency,
+    'type': 'object',
+    'properties': {
+        'locale': {
+            'type': 'string',
+            'description': 'TODO',
+        },
+        'locale_configfile': {
+            'type': 'string',
+            'description': 'TODO',
+        },
+    }
```

Comment:
I think you are missing an 'additionalProperties': False
which makes sure we don't allow anything besides locale/locale_configfile in the config dict.

Reply:
I don't think so in this case, because that rejects any top-level keys other than "locale" and "locale_configfile", meaning if I add it then this file:

```yaml
#cloud-config
locale: ar
locale_configfile: "some string"
runcmd: ["foo"]
```

gives this error:

```
Cloud config schema errors: : Additional properties are not allowed ('runcmd' was unexpected)
```

(I think you're thinking of a case like [`cc_ubuntu_drivers.py`](https://github.com/canonical/cloud-init/blob/master/cloudinit/config/cc_ubuntu_drivers.py#L36) where the _value_ of a config key is an object; in those cases, the schema for that config key _can_ define the full set of keys that the object should take.)

Reply:
Ahh right you are correct. Since these are top-level properties and not nested objects, we don't provide the strict schema definition preventing additional keys. We may want tighten that in the GLOBAL_SCHEMA definition once we have full coverage, but this is not something for individual config module schema definitions to provide.